dtbase/services/base.py

```diff
@@ -0,0 +1,218 @@
+"""
+Base class for all services used in the application.
+"""
+from typing import Any, List, Optional
+
+from requests import Response
+
+from dtbase.core.constants import (
+    DEFAULT_USER_EMAIL,
+    DEFAULT_USER_PASS,
+)
+from dtbase.core.utils import auth_backend_call, log_rest_response, login
+
+
+class BaseService:
+    """
+    Base class for all services. Users inherit this class to write custom modelling and
+    data ingress services. The user should implement the following methods:
+    """
+
+    def __init__(self) -> None:
+        self.access_token = None
+        self.service_type = None
+
+    def _backend_login(self, username: str, password: Optional[str]) -> None:
+        """
+        Sets the access token using login credentials.
+
+        Raises BackendCallError if the login fails.
+        """
+        self.access_token = login(username, password)[0]
+
+    def get_service_data(self) -> None:
```

Comment:
I wonder if the return type annotation here should be what the method should return once it's been implemented. So something like `list[tuple[str, dict]]`. That would help document how this method is expected to be used.